System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -18,7 +18,7 @@ public class MockDirectory : DirectoryBase
 
         private string currentDirectory;
 
-        public MockDirectory(IMockFileDataAccessor mockFileDataAccessor, FileBase fileBase, string currentDirectory)
+        public MockDirectory(IMockFileDataAccessor mockFileDataAccessor, FileBase fileBase, string currentDirectory) : base(mockFileDataAccessor?.FileSystem)
```

Comment:
The reason the above feels a bit icky is that you end up with these `null` checks which don't make all that much sense. The alternative here is obviously to pass in both the `IMockFileDataAccessor` and the `IFileSystem` to all these. Would people prefer we do that instead?